main.js

```diff
@@ -0,0 +1,446 @@
+    const forecastDays = {};
+    let dayCount = 0;
+
+    data.list.forEach(item => {
+        const date = new Date(item.dt * 1000);
+        const dayString = date.toLocaleDateString('en-US', { weekday: 'short', month: 'short', day: 'numeric' });
+
+        if (dayCount < 8 && !forecastDays[dayString]) {
+            forecastDays[dayString] = {
+                temp: Math.round(item.main.temp),
+                icon: weatherIcons[item.weather[0].main] || 'default.png',
+                description: item.weather[0].description || "No description available"
+            };
+            dayCount++;
+        }
+    });
+
+    // Add forecast data to the table
+    Object.keys(forecastDays).forEach(day => {
+        const dayData = forecastDays[day];
+        console.log(dayData);
+        const row = document.createElement('tr');
+        row.innerHTML = `
+            <td>${day}</td>
+            <td class="forecast-icon"><img src="icons/${dayData.icon}" alt="Weather icon" >${dayData.temp}°C</td>
+            <td>${dayData.description}</td>
+        `;
+        tableBody.appendChild(row);
+    });
+
+    if (forecastModal && modalBackdrop) {
+        forecastModal.style.display = 'block';
+        modalBackdrop.style.display = 'block';
+    } else {
+        console.error("Modal or backdrop not found.");
+    }
+
+    modalBackdrop.addEventListener('click', () => {
+        forecastModal.style.display = 'none';
+        modalBackdrop.style.display = 'none';
+    });
+}
+
+searchInput.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+        searchButton.click();
+    }
+})
+
+function fetchWeatherByCoordinates(lat, lon) {
+    fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${apiKey}&units=metric`)
+        .then(response => {
+            if (!response.ok) {
+                throw new Error('Location not found');
+            }
+            return response.json();
+        })
+        .then(data => {
+            console.log(data);
+            if (data.main && data.weather && data.weather[0]) {
+                displayWeatherData(data);
+            } else {
+                console.error('Weather data is missing or incomplete:', data);
+                errorMessage.textContent = 'Weather data is missing or incomplete.';
+                errorMessage.style.display = 'block';
+                loadingSpinner.style.display = 'none';
+            }
+        })
+        .catch(error => {
+            console.error('Error fetching weather data by coordinates:', error);
+            errorMessage.textContent = 'Unable to fetch weather data. Please try again later.';
+            errorMessage.style.display = 'block';
+            loadingSpinner.style.display = 'none';
+        });
+}
+
+function getCurrentLocationWeather() {
+    if (navigator.geolocation) {
+        loadingSpinner.style.display = 'block';
+        navigator.geolocation.getCurrentPosition(function (position) {
```

Comment:
navigator.geolocation.getCurrentPositio is duplicate code, create a separate function, which will return lat and long. and use that function